src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineWrappingInBlockCheck.java

```diff
@@ -0,0 +1,440 @@
+                // Special handling for the "else if" case.
+                final DetailAST candidate = ast.getFirstChild();
+                if (isRightCurlyChildOf(candidate)) {
+                    leftBrace = candidate;
+                }
+                break;
+            case TokenTypes.SLIST:
+                // The scope block is a SLIST with another SLIST as the parent.
+                if (ast.getParent().getType() == TokenTypes.SLIST) {
+                    leftBrace = ast;
+                }
+                break;
+            case TokenTypes.LITERAL_SWITCH:
+                // The Switch does not have SLIST unlike the others.
+                leftBrace = ast.findFirstToken(TokenTypes.LCURLY);
+                break;
+            case TokenTypes.CASE_GROUP:
+                // The CASE_GROUP is the parent of 'case', 'default' and SLIST parent.
+                leftBrace = ast.getLastChild().findFirstToken(TokenTypes.SLIST);
+                break;
+            case TokenTypes.LITERAL_CASE:
+            case TokenTypes.LITERAL_DEFAULT:
+                // The 'case' and 'default' are siblings for SLIST parent.
+                leftBrace = ast.getParent().getLastChild().findFirstToken(TokenTypes.SLIST);
+                break;
+            default:
+                // All the rest: for/while/if/STATIC_INIT,lambda,etc.
+                leftBrace = ast.findFirstToken(TokenTypes.SLIST);
+                break;
+        }
+
+        if (leftBrace != null) {
+            verify(leftBrace);
+        }
+    }
+
+    /**
+     * Verifies that a specified code block is wrapped correctly according to the policy.
+     * @param leftBrace token for left curly brace
+     */
+    private void verify(DetailAST leftBrace) {
+        if (isRightCurlyChildOf(leftBrace)) {
+            // For the SLIST and the ARRAY_INIT the right curly brace is the last child of
+            // the left curly brace.
+            final DetailAST rightBrace = leftBrace.getLastChild();
+            // This check does not validate empty blocks.
+            if (leftBrace.getFirstChild() != rightBrace) {
+                verifyTop(leftBrace);
+                verifyBottom(rightBrace);
+            }
+        }
+        else {
+            // For the OBJBLOCK/SWITCH the left curly brace is the first child and
+            // the right curly brace is the last child of their common parent.
+            final DetailAST rightBrace = leftBrace.getParent().getLastChild();
+            // This check does not validate empty blocks.
+            if (leftBrace.getNextSibling() != rightBrace) {
+                verifyTop(leftBrace);
+                verifyBottom(rightBrace);
+            }
+        }
+    }
+
+    /**
+     * Verifies that a specified code block matches the policy for the top part of the block.
+     * @param brace token for left curly brace
+     */
+    private void verifyTop(DetailAST brace) {
+
+        if (topSeparator != BlockWrapOption.EMPTY_LINE_ALLOWED) {
```

Comment:
this should be in `visitToken` and skip execution completely is it is defined by user.

Reply:
The execution should be skipped completely only if both separators are optional.
In this case the check has no effect at all and should be removed from the config.
As you noted before.

Reply:
looks like one more point for having two Checks for each brace.